### DISCUSSION TITLE
CMI Warnings Lifetime

### DISCUSSION THREAD

**Discussion by Albert481** [Zrips/CMI#4262] [Q&A]
**Question: **
How does CMI keep track of the warning lifetime on the user?

Becos database still stores the warning but in ingame /warning <player> shows empty

**Comment by CloudeLecaw** [comment]
Normaly when you give a warning u define a time in config after that time the warning gets removed. If yoir using mysql then its stored there. If your using sqlite its stored in the db file.  Wenn player write warnings then he should het a list with all his remaining warnings. You could take a look in your langfile if warning gets listed ingame in correct format.

**Comment by Albert481** [comment]
I did not change anything in the langfiles. The /warning shows empty but the database still stores the warning, date of issue and the issuer name  **after the warning has expired.**

**Comment by CloudeLecaw** [comment]
this is normal. you can always "history" the warning that way

if you want to edit this u need to use


> editwarnings (playerName/clearall) (clear)
> Edit players warnings

/warnings just shows activ warnings

**Comment by Albert481** [comment]
The reason for this question because I have a [panel](https://panel.blockstackers.xyz/warnings) that tracks player warnings on my network. Even though their warnings have expired ingame, the warnings are still stored hence does not reflect accurate results with the unnecessary data.
